### ISSUE TITLE
slower than cnn

### ISSUE THREAD

**Issue by sci-study-storage** [traveller59/spconv#457]
Hi Thanks for this excellent work!
I would like to ask a question regarding the speed of training and prediction.
I'm assuming that sparse cnn is more computationally efficient and should be faster than the cnn. But in my neural network, cnn shows faster results both for training and prediction. I think there are two possibilities in my mind. 

First, the sparse cnn including spconv, does not always show better performance than cnn in some datasets.
Second, I have not fully exploited the best performance of the spconv. 

Could you please give any comments regarding my thoughts?

FYI, I'm attaching my architecture which is comparable VGG for CNN. 

class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.net = spconv.SparseSequential(            
            spconv.SubMConv3d(1, 32, 3),
            nn.BatchNorm1d(32),
            nn.ReLU(),
            spconv.SubMConv3d(32, 32, 3),
            nn.BatchNorm1d(32),
            nn.ReLU(),            
            spconv.SparseMaxPool3d(2, 2, 0, 1),
            
            spconv.SubMConv3d(32, 64, 3),
            nn.BatchNorm1d(64),
            nn.ReLU(),
            spconv.SubMConv3d(64, 64, 3),
            nn.BatchNorm1d(64),
            nn.ReLU(),            
            spconv.SparseMaxPool3d(2, 2, 0, 1),
            
            spconv.SubMConv3d(64, 128, 3),
            nn.BatchNorm1d(128),
            nn.ReLU(),
            spconv.SubMConv3d(128, 128, 3),
            nn.BatchNorm1d(128),
            nn.ReLU(),
            spconv.SubMConv3d(128, 128, 3),
            nn.BatchNorm1d(128),
            nn.ReLU(),            
            spconv.SparseMaxPool3d(2, 2, 0, 1),            

            spconv.SubMConv3d(128, 256, 3),
            nn.BatchNorm1d(256),
            nn.ReLU(),
            spconv.SubMConv3d(256, 256, 3),
            nn.BatchNorm1d(256),
            nn.ReLU(),
            spconv.SubMConv3d(256, 256, 3),
            nn.BatchNorm1d(256),
            nn.ReLU(),            
            spconv.SparseMaxPool3d(2, 2, 0, 1),             
            
            spconv.ToDense(),
        )
        
        self.regressor = nn.Sequential(
            nn.Linear(256*4*4*4, 2048),
            nn.ReLU(True),
            nn.Dropout(),
            nn.Linear(2048, 2048),
            nn.ReLU(True),
            nn.Dropout(),
            nn.Linear(2048, 1),
        )

    def forward(self, x: torch.Tensor):
        torch_tensor_dense_to_sp = x[:,0,:,:].to_sparse() 
        features = torch_tensor_dense_to_sp.values().view(-1, 1) # your features with shape [N, num_channels]
        indices = torch_tensor_dense_to_sp.indices().permute(1,0).contiguous().int()
        spatial_shape = x[:,0,:,:].shape[1:]
        batch = x.shape[0]

        x_sp = spconv.SparseConvTensor(features, indices, spatial_shape, batch)
        # create SparseConvTensor manually: see SparseConvTensor.from_dense
        x = self.net(x_sp)
        x = torch.flatten(x, 1)
        x = self.regressor(x)
        return x

I appreciate your comments in advance!

**Comment by qifeng22** [comment]
Do you compare the memory? In my acknowledge, the purpose of sparse conv is saving the memory. @jh-chung1
